### ISSUE TITLE
Segmentation fault due to double free caused by Application::restart()

### ISSUE THREAD

**Issue by yan12125** [keepassxreboot/keepassxc#4916]
## Overview
In both `Application::restart()` and `Application::~Application()`, `m_lockFile` is unlocked and deleted, causing double free and thus segmentation fault if the application is restarted.

gdb stack trace:
```
#0  0x00007ffff6bfcbed in QLockFile::unlock() () at /usr/lib/libQt5Core.so.5
#1  0x00005555555dbe02 in Application::~Application() (this=0x7fffffffddb0, __in_chrg=<optimized out>) at /build/keepassxc-git/src/keepassxc/src/gui/Application.cpp:140
#2  0x00005555555c386b in main(int, char**) (argc=<optimized out>, argv=0x7fffffffdd50) at /build/keepassxc-git/src/keepassxc/src/main.cpp:95
```

Relevant lines from Valgrind:
```
==5464== Invalid read of size 8
==5464==    at 0x5B4BBDA: QLockFile::unlock() (in /usr/lib/libQt5Core.so.5.15.0)
==5464==    by 0x18FE01: Application::~Application() (Application.cpp:140)
==5464==    by 0x17786A: main (main.cpp:56)
==5464==  Address 0xd1c6880 is 0 bytes inside a block of size 8 free'd
==5464==    at 0x483B08B: operator delete(void*, unsigned long) (vg_replace_malloc.c:593)
==5464==    by 0x18EEB1: Application::restart() (Application.cpp:325)
==5464==    by 0x4C7FC7D: QWidget::event(QEvent*) (in /usr/lib/libQt5Widgets.so.5.15.0)
==5464==    by 0x4C3E701: QApplicationPrivate::notify_helper(QObject*, QEvent*) (in /usr/lib/libQt5Widgets.so.5.15.0)
==5464==    by 0x5BB9699: QCoreApplication::notifyInternal2(QObject*, QEvent*) (in /usr/lib/libQt5Core.so.5.15.0)
==5464==    by 0x4C7ABDA: QWidgetPrivate::close_helper(QWidgetPrivate::CloseMode) (in /usr/lib/libQt5Widgets.so.5.15.0)
==5464==    by 0x192552: operator() (MainWindow.cpp:1692)
==5464==    by 0x192552: call (qobjectdefs_impl.h:146)
==5464==    by 0x192552: call<QtPrivate::List<bool>, void> (qobjectdefs_impl.h:256)
==5464==    by 0x192552: QtPrivate::QFunctorSlotObject<MainWindow::initViewMenu()::{lambda(bool)#2}, 1, QtPrivate::List<bool>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) (qobjectdefs_impl.h:443)
==5464==    by 0x5BF0905: ??? (in /usr/lib/libQt5Core.so.5.15.0)
==5464==    by 0x4C37F82: QAction::toggled(bool) (in /usr/lib/libQt5Widgets.so.5.15.0)
==5464==    by 0x4C3A6D4: QAction::setChecked(bool) (in /usr/lib/libQt5Widgets.so.5.15.0)
==5464==    by 0x4C3A7D5: QAction::activate(QAction::ActionEvent) (in /usr/lib/libQt5Widgets.so.5.15.0)
==5464==    by 0x4DBBF1A: ??? (in /usr/lib/libQt5Widgets.so.5.15.0)
==5464==  Block was alloc'd at
==5464==    at 0x4839DEF: operator new(unsigned long) (vg_replace_malloc.c:342)
==5464==    by 0x18F7E6: Application::Application(int&, char**) (Application.cpp:88)
==5464==    by 0x1772A9: main (main.cpp:56)
```

Looks like the latest Qt is more sensitive to application bugs than previous Qt versions :thinking: 

## Steps to Reproduce
1. Check or uncheck View -> Compact Mode

## Expected Behavior
Application is restarted with the specified mode

## Actual Behavior
Application is restarted, but the previous instance crashed with segmentation fault.

## Context

KeePassXC - Version 2.6.0-snapshot
Build Type: Snapshot
Revision: 05a0df7

Qt 5.15.0
Debugging mode is enabled.

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 5.7.6-arch1-1

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
 libgcrypt 1.8.5

Operating System: Arch Linux
Desktop Env: LXQt
Windowing System: X11

**Comment by droidmonkey** [comment]
Oof good catch!

**Comment by droidmonkey** [comment]
Fixed in ef7a4288ae09bce28678fd989de40a72cee1a3bf
